src/config.rs

```diff
@@ -141,4 +142,40 @@ impl ParserConfig {
         self.transform_block = Some(Rc::new(callback));
         self
     }
+
+    /// Allows unmatched tag pairs where the element opens with a block and
+    /// the close tag matches a specified wildcard. For example
+    ///
+    /// ```rust
+    /// use quote::quote;
+    /// use rstml::{Parser, ParserConfig, node::{Node, NodeElement}};
+    /// use syn::{Expr, ExprRange, RangeLimits, Stmt};
+    ///
+    /// let config = ParserConfig::new().block_element_close_wildcard(|close_tag| {
+    ///     // Construct a sample of what a wildcard block element close should be
+    ///     let sample_close = quote! { <{}></_> };
+    ///     let parsed = Parser::new(ParserConfig::new().block_element_close_wildcard(|_| true))
+    ///         .parse_simple(sample_close);
+    ///     let Ok([Node::Element(NodeElement {
+    ///         close_tag: Some(sample_close),
+    ///         ..
+    ///     })]) = parsed.as_deref() else {
+    ///         unreachable!();
+    ///     };
+    ///     close_tag.name == sample_close.name
+    /// });
+    ///
+    /// let tokens = quote! {
+    ///     <{"OpenTag"}>{"Content"}</_>
+    /// };
+    ///
+    /// Parser::new(config).parse_simple(tokens).unwrap();
+    /// ```
+    pub fn block_element_close_wildcard<F>(mut self, predicate: F) -> Self
```

Comment:
I like the idea of PR.
What do you think about making it generic, and instead of allowing wildcard for "block" element, also allow to use wildcard in regular tags too?

If you accept it then method should be renamed, and closure should also receive `OpenTag`. And it would be a good idea to add getters for `NodeName` (`is_wildcard`/`is_block`/`is_dashed`).